pricegraph/src/orderbook/scalar.rs

```diff
@@ -169,3 +167,188 @@ fn assert_strictly_positive_and_finite(value: f64) -> f64 {
     debug_assert!(num::is_strictly_positive_and_finite(value));
     value
 }
+
```

Comment:
This is some very good documentation. Would it make sense to pull the weight into its own module?